interpreter/astwalker.go

```diff
@@ -36,26 +36,28 @@ const (
 func WalkExpr(expression *expr.Expr,
 	metadata Metadata,
 	dispatcher Dispatcher,
-	state MutableEvalState) []Instruction {
+	state MutableEvalState) ([]Instruction, map[int64]int64) {
```

Comment:
Just a thought, but would it make more sense to build this mapping in the MutableEvalState? Since this state should be communicable across process boundaries, having more information here beyond a simple map from runtime id to value is probably useful. I believe it would also simplify the prune contract to:
`PruneExpr(expr, state)`